### ISSUE TITLE
Gridin

### ISSUE THREAD

**Issue by WangYiKang0** [dealii/dealii#17122]
I was using gmsh to draw a model for fsi and got an error when using grid_in for input. How can I solve such a problem.

```
terminate called after throwing an instance of 'dealii::StandardExceptions::ExcMessage'
  what():  
--------------------------------------------------------
An error occurred in line <2594> of file </public3/home/sc52180/code/dealii-9.5.1/source/grid/tria.cc> in function
    static void dealii::internal::TriangulationImplementation::Implementation::process_subcelldata(const dealii::internal::TriangulationImplementation::CRS<T>&, dealii::internal::TriangulationImplementation::TriaObjects&, const std::vector<dealii::CellData<dim> >&, const std::vector<dealii::Point<spacedim> >&) [with int structdim = 2; int spacedim = 3; T = unsigned int]
The violated condition was: 
    boundary_id != numbers::internal_face_boundary_id
Additional information: 
    The input arguments for creating a triangulation specified a boundary
    id for an internal face. This is not allowed.
    
    The object in question has vertex indices 2,187,2185,2208, which are
    located at positions (-13.2811 -5.08123 6.5)(-10.368 -6.89958
    6.5)(-12.5063 -7.81241 5.63899)(-10.7098 -7.8423 6.04588).
```

Here only a part of the mesh is shown (the outer contour also generates the mesh),the error should occur at the interface of the two bodies, but I don't know how to fix it. This model will be used for fluid-solid coupling, where the interior is the fluid domain and the contours are the solid domain.
![捕获](https://github.com/dealii/dealii/assets/149748911/00d086f4-25ec-4ab8-a087-4b69409b50eb)

**Comment by drwells** [comment]
I don't think it is possible to answer such a question without more context. Was this error from `GridIn::read_msh()`, or did you try to merge two Triangulations?

If it is the former - deal.II cannot load meshes which set any kind of boundary id on internal faces. You will have to clean up your data before loading it into deal.II.

There are a lot more people working on these kinds of problems on the mailing list than in the issue tracker - you might get a better response there than here for this kind of problem. Would you prefer to move the discussion there?

**Comment by WangYiKang0** [comment]
您的邮件已收到

**Comment by WangYiKang0** [comment]
I think it happens in GridIn::read_msh(). I've answered this question on the mailing list but the response I got was not that the problem's been solved!

**Comment by drwells** [comment]
I don't think that there is any solution to this problem besides fixing the input `.geo` file to not define boundaries inside the structure.

**Comment by WangYiKang0** [comment]
I tried to do this, but it was still wrong. I draw the interface of the fluid-solid coupling without imposing any physical labels. It still doesn't pass the Gridin

**Comment by WangYiKang0** [comment]
您的邮件已收到

**Comment by bangerth** [comment]
If anyone is curious what the last comment means: "Your email has been received" :-)

**Comment by drwells** [comment]
@WangYiKang0 did you get this working?

**Comment by WangYiKang0** [comment]
您的邮件已收到

**Comment by drwells** [comment]
I don't think there's anything else we can do here without a minimum working example.

**Comment by bangerth** [comment]
For reference, the Chinese text above translates to "Your email has been received".
